sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -26,8 +28,17 @@ import org.apache.spark.sql.catalyst.TableIdentifier
 @Experimental
 trait LookupCatalog {
 
+  protected def defaultCatalogName: Option[String] = None
   protected def lookupCatalog(name: String): CatalogPlugin
 
+  lazy val defaultCatalog: Option[CatalogPlugin] = {
+    defaultCatalogName.flatMap(name => Try(lookupCatalog(name)).toOption)
+  }
+
+  lazy val sessionCatalog: Option[CatalogPlugin] = {
```

Comment:
Don't think sessionCatalog belong to the abstraction of LookupCatalog interface.

Reply:
I can move it out, but it requires exposing `lookupCatalog` for `DataSourceResolution` to get the session catalog. I would rather keep it here to avoid calls to `lookupCatalog("session")` in other places.